src/test/run-pass/or-patterns.rs

```diff
@@ -0,0 +1,45 @@
+#![feature(or_patterns)]
```

Comment:
Btw, please move any and all new tests into `ui/or-patterns` for better semantic categorization :)

Reply:
Good suggestion!